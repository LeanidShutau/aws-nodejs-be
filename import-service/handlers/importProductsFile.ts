import { APIGatewayProxyHandler } from 'aws-lambda';
import AWS from 'aws-sdk';
import 'source-map-support/register';

const headers = {
  'Access-Control-Allow-Origin': '*',
};

export const importProductsFile: APIGatewayProxyHandler = async (event, _context) => {
  try {
    console.log('queryStringParameters: ', event.queryStringParameters);
    const { name } = event.queryStringParameters || {};
    const path = `uploaded/${name}`;

    const s3 = new AWS.S3({ region: 'eu-west-1' });
    const params = {
      Bucket: 'lshutau-rs-uploaded',
      Key: path,
      Expires: 60,
      ContentType: 'text/csv',
    };

    const url = await new Promise<string>((resolve, reject) => {
      s3.getSignedUrl('putObject', params, (err, url) => {
        if (err) {
          return reject(err);
        }

        resolve(url);
      });
    });

    return {
      statusCode: 200,
      body: url,
      headers,
    };
  } catch (err) {
    return {
      statusCode: 500,
      body: JSON.stringify(err, null, 2),
      headers,
    };
  }
}
