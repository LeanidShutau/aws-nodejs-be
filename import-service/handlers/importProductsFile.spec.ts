import { APIGatewayProxyResult } from 'aws-lambda';
import * as AWSMock from 'aws-sdk-mock';

import { importProductsFile } from './importProductsFile';

describe('importProductsFile', () => {
  it('should return signedUrl', async () => {
    AWSMock.mock('S3', 'getSignedUrl', (_, __, cb) => {
      cb(null, 'https://signedurl.com');
    });
    const signedUrl = await importProductsFile({
      queryStringParameters: {
        name: 'file.csv'
      }
    } as any, null, null) as unknown as APIGatewayProxyResult;

    expect(signedUrl.body).toMatchSnapshot();
  });
})
