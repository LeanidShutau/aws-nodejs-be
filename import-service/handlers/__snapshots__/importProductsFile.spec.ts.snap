// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importProductsFile should return signedUrl 1`] = `"{}"`;
